ToDo-Garden/ToDoGardenUI/Sources/ToDoGardenUIComponent/UIButton+AddButton.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  UIButton+addButton.swift
+//
+//
+//  Created by Wood on 3/4/24.
+//
+
+import ToDoGardenUIResource
+import UIKit.UIButton
+
+extension UIButton {
+  public func addButtonStyle(with title: String) {
+    self.setupUIAppearance(with: title)
+  }
+}
+
+extension UIButton {
+  private func setupUIAppearance(with title: String) {
+    AddButtonStyle.apply(for: self, with: title)
+  }
+}
+
+private enum AddButtonStyle {
+  fileprivate static func apply(for button: UIButton, with title: String) {
+    AddButtonStyle.setupImage(for: button)
+    AddButtonStyle.setupTintColorWhenStateChanged(for: button)
+    AddButtonStyle.setupTitle(for: button, with: title)
+  }
+}
+
+extension AddButtonStyle {
+  private static func setupImage(for button: UIButton) {
+    let renderingMode = UIImage.RenderingMode.alwaysTemplate
+    let imagePadding: CGFloat = 4.0
+    var configuration = UIButton.Configuration.plain()
+    configuration.image = UIImage.addButton.withRenderingMode(renderingMode)
+    configuration.imagePadding = imagePadding
+    configuration.imagePlacement = NSDirectionalRectEdge.leading
+    button.configuration = configuration
+  }
```

Comment:
```suggestion
  fileprivate static func apply(
    for button: UIButton,
    with title: String
  ) {
    AddButtonStyle.applyConfiguration(
      for: button,
      with: UIButton.Configuration.plain()
    )
    AddButtonStyle.configureImage(for: button)
    AddButtonStyle.setImage(for: button)
    AddButtonStyle.setupTintColorWhenStateChanged(for: button)
    AddButtonStyle.setupTitle(for: button, with: title)
  }
}

extension AddButtonStyle {
private static func applyConfiguration(
  for button: UIButton,
  with configuration: UIButton.Configuration
) {
  button.configuration = configuration
}
	
private static func configureImage(for button: UIButton) {
  let imagePadding: CGFloat = 4.0
  button.configuration?.imagePadding = imagePadding
  button.configuration?.imagePlacement = NSDirectionalRectEdge.leading
}
	
private static func setImage(for button: UIButton) {
  let renderingMode = UIImage.RenderingMode.alwaysTemplate
  let addButtonImage = UIImage.addButton.withRenderingMode(renderingMode)
  button.setImage(addButtonImage, for: .normal)
}

```

위 코드처럼 하나의 함수가 하나의 일만 하도록 분리해보는건 어떨까요?

Reply:
직접 코드까지.. 정말 감사합니다!! 단계를 나눠서 반영해보았습니다
> https://github.com/ToDo-Garden/ToDo-Garden/pull/42/commits/c86277b432bbb5af7736d232c8d99afdf08a2c0d